analyzers/src/SonarAnalyzer.Common/Rules/Utilities/AnalysisWarningAnalyzerBase.cs

```diff
@@ -28,29 +28,45 @@ public abstract class AnalysisWarningAnalyzerBase : UtilityAnalyzerBase
     private const string DiagnosticId = "S9999-warning";
     private const string Title = "Analysis Warning generator";
 
-    protected virtual int MinimalSupportedRoslynVersion => RoslynHelper.MinimalSupportedMajorVersion;   // For testing
+    protected virtual int UnsupportedVersionVS2017 => RoslynHelper.UnsupportedVersionVS2017; // For testing
+    protected virtual int MinimalSupportedRoslynVersion => RoslynHelper.MinimalSupportedMajorVersion; // For testing
 
     protected AnalysisWarningAnalyzerBase() : base(DiagnosticId, Title) { }
 
     protected sealed override void Initialize(SonarAnalysisContext context) =>
         context.RegisterCompilationAction(c =>
             {
                 var parameter = ReadParameters(c);
-                if (parameter.IsAnalyzerEnabled && !RoslynHelper.IsRoslynCfgSupported(MinimalSupportedRoslynVersion))     // MsBuild 15 is bound with Roslyn 2.x, where Roslyn CFG is not available.
+                if (!parameter.IsAnalyzerEnabled || parameter.OutPath is null)
                 {
-                    // This can be removed after we bump Microsoft.CodeAnalysis references to 3.0 or higher.
-                    var path = Path.GetFullPath(Path.Combine(parameter.OutPath, "../../AnalysisWarnings.MsBuild.json"));
-                    if (!File.Exists(path))
+                    return;
+                }
+
+                var path = Path.GetFullPath(Path.Combine(parameter.OutPath, "../../AnalysisWarnings.MsBuild.json"));
+                if (!File.Exists(path))
+                {
+                    // This can be removed after we bump Microsoft.CodeAnalysis references to 2.0 or higher. MsBuild 14 is bound with Roslyn 1.x.
+                    if (!RoslynHelper.IsRoslynCfgSupported(UnsupportedVersionVS2017)) // Roslyn CFG is not available.
```

Comment:
The check we want to do is not related to CFG at all. Due to this, the code is misleading.

I think we should:
- extract a method in `RoslynHelper`; it can be called `IsVersionLessThan` or `IsVersionGreaterThan`, which can do the version check.
- use that method here 
- reuse that method in `IsRoslynCfgSupported`
- remove the comment about Roslyn CFG